lib/pipeline.js

```diff
@@ -1806,7 +1828,24 @@ class PipelineModel extends BaseModel {
      * collections, and remove the pipeline itself
      * @return {Promise}        Resolves to null if removed successfully
      */
-    remove() {
+    async remove() {
+        const latestBuilds = await this.getBuilds();
+
+        if (latestBuilds.some(b => ['QUEUED', 'RUNNING', 'BLOCKED'].includes(b.status))) {
```

Comment:
Should we include all the non-terminal statuses?
Ex: `FROZEN`, `CREATED`
These may get transitioned to `QUEUED` or `RUNNING` when the deletion is in progress.

Reply:
I excluded the `FROZEN` and `CREATED` statuses for the following reasons:

- I think `FROZEN` is not running status, it is a type of scheduled build. Therefore we can block the execution of scheduled builds at the `POST /events` endpoint.
- Sometime `CREATED` builds remain.
  - For instance, the build `c`, which has `requires: [ a, b ]`, is created after the build `a`. However if the job `b` is disabled, the build `c` is not started and not deleted.